lib/scarlet/plugin.rb

```diff
@@ -23,10 +24,18 @@ def helper(*args, &block)
   end
 
   class Context
-    def initialize *objs
+    attr_accessor :event
+
+    def initialize event, helpers, *objs
+      @event = event
+      extend(helpers)
```

Comment:
Not sure what the performance impact of this is, in the worst case we could generate these on plugin initialization, and store a context class per plugin (then just use the specific one to generate a new context).